include/SFML/Network/Ftp.hpp

```diff
@@ -505,7 +506,7 @@ class SFML_NETWORK_API Ftp
     /// \see download
     ///
     ////////////////////////////////////////////////////////////
-    [[nodiscard]] Response upload(const std::string& localFile, const std::string& remotePath, TransferMode mode = Binary, bool append = false);
+    [[nodiscard]] Response upload(const std::filesystem::path& localFile, const std::filesystem::path& remotePath, TransferMode mode = Binary, bool append = false);
```

Comment:
Are these *always* valid paths?

Reply:
What do you mean?

Reply:
I don't know exactly, just wondering if the FTP protocol enforces normal paths or if there are also cases where you might not have "normal" paths but just some random strings 🤔

Reply:
I see. `std::filesystem::path` does normalize the path to remove unnecessary bits like `../` or multiple consecutive backslashes since those things add no extra info to a path. Maybe that removal of characters could break something here. I'm not sure.

Reply:
Has any research/testing been done here? Maybe worthwhile revert the changes here to merge the PR

Reply:
No I haven't tested this particular snippet. I'll revert this for now.

Reply:
On second inspection, `std::filesystem::path::path` does not appear to modify the string you give it upon construction. It merely stores that string. Any manipulation of that path happens later upon request when calling functions like [`std::filesystem::canonical`](https://en.cppreference.com/w/cpp/filesystem/canonical). I believe it's safe to use `std::filesystem` here.